java/java.source/src/org/netbeans/modules/java/JavaNode.java

```diff
@@ -429,6 +385,147 @@ public String getValue() {
+            }
+
+            private static boolean isCompatible(SpecificationVersion jdkVerison) {
+                try {
+                    return Integer.parseInt(jdkVerison.toString()) >= 11;
+                } catch (NumberFormatException ignore) {}
+                return false;
+            }
+
+            @Override
+            public void setAsText(String text) throws IllegalArgumentException {
+                jdkCombo.setSelectedItem(text);
+            }
+
+            @Override
+            public void setValue(Object value) {
+                jdkCombo.setSelectedItem(value);
+                super.setValue(value);
+            }
+
+            @Override
+            public Component getCustomEditor() {
+                return jdkCombo;
+            }
+
+            @Override
+            public boolean supportsCustomEditor() {
+                return true;
+            }
+        }
+
+    }
+
+    private static final class RunFileArgumentsProperty extends PropertySupport.ReadWrite<String> {
+
+        private final DataObject dObj;
+
+        public RunFileArgumentsProperty(DataObject dObj) {
+            super("runFileArguments", String.class, // NOI18N
+                  "Arguments",
+                  "Arguments passed to the main method while running the file.");
+            this.dObj = dObj;
+        }
+
+        @Override
+        public String getValue() {
+            return dObj.getPrimaryFile().getAttribute(FILE_ARGUMENTS) instanceof String args ? args : "";
+        }
+
+        @Override
+        public void setValue(String o) {
+            try {
+                dObj.getPrimaryFile().setAttribute(FILE_ARGUMENTS, o);
+            } catch (IOException ex) {
+                LOG.log(Level.WARNING, "Java File does not exist : {0}", dObj.getPrimaryFile().getName()); //NOI18N
+            }
+        }
+    }
+
+    private static class RunFileVMOptionsProperty extends PropertySupport.ReadWrite<String> {
+
+        private final DataObject dObj;
+
+        public RunFileVMOptionsProperty(DataObject dObj) {
+            super("runFileVMOptions", String.class, // NOI18N
+                  "VM Options",
+                  "VM Options to be considered while running the file.");
+            this.dObj = dObj;
+        }
+
+        @Override
+        public String getValue() {
+            return dObj.getPrimaryFile().getAttribute(FILE_VM_OPTIONS) instanceof String opts ? opts : "";
+        }
+
+        @Override
+        public void setValue(String o) {
+            try {
+                dObj.getPrimaryFile().setAttribute(FILE_VM_OPTIONS, o);
+                // what is this for?
+                Source s = Source.create(dObj.getPrimaryFile());
+                ModificationResult result = ModificationResult.runModificationTask(List.of(s), new UserTask() {
+                    @Override public void run(ResultIterator resultIterator) {}
+                });
+                result.commit();
```

Comment:
i am not sure what this does, if it doesn't do anything, the two read/write properties can be refactored to one class, since this is the only difference between the two

Reply:
Well Squash them to one. One is the VM args, and the other is the args for the java file to be executed.

Reply:
done